Re-organizing and editing text input docs (#918)
hashicorp/design-system: website/docs/components/form/text-input/partials/guidelines/guidelines.md

## Usage

### When to use

- As a form element that provides users with a way to read, input, or edit data in a freeform way.

### When not to use

- If needing a multi-line text input, use [Textarea](/components/form/textarea/overview)
- If needing to allow the user to make a selection from a predetermined list of options, use [Checkbox](/components/form/checkbox/overview), [Radio button](/components/form/radio/overview), or [Select](/components/form/select/overview).

### Required and optional

> For complex forms, indicate **required** fields. This is the most explicit and transparent method and ensures users don't have to make assumptions. Read more about best practices for [marking required fields in forms](https://www.nngroup.com/articles/required-fields/).

Reviewer: ```suggestion
For complex forms, indicate **required** fields. This is the most explicit and transparent method and ensures users don’t have to make assumptions. Read more about best practices for [marking required fields in forms](https://www.nngroup.com/articles/required-fields/).
```

<Hds::Form::TextInput::Field @type="text" @isRequired={{true}} @width="300px" as |F|>
  <F.Label>Label</F.Label>
</Hds::Form::TextInput::Field>

For shorter, simpler forms (ie. login/signup and feedback requests), indicate **optional** fields instead.

<Hds::Form::TextInput::Field @type="text" @isOptional={{true}} @width="300px" as |F|>
  <F.Label>Label</F.Label>
</Hds::Form::TextInput::Field>

### Readonly, disabled, and hidden fields

Readonly and disabled fields are similar in that the value is not editable by the user. Where they differ is in what happens to the data upon form submission. Data in readonly fields gets passed when the form is submitted, while data in disabled fields does not. 

<Hds::Form::TextInput::Field @type="text" @width="300px" readonly as |F|>
  <F.Label>Readonly</F.Label>
</Hds::Form::TextInput::Field>

<Hds::Form::TextInput::Field @type="text" @width="300px" disabled as |F|>
  <F.Label>Disabled</F.Label>
</Hds::Form::TextInput::Field>

We recommend using a hidden field, if the data needs to be submitted but not reviewed by the user first. 

Use these fields sparingly.

> ### Validation

Reviewer: ```suggestion
### Validation

```
An empty line should follow each heading.

!!! Warning

While we provide the structure and visual consistency for validation, the messaging and functionality are the responsibility of the product teams.
!!!

> #### Types of validation

Reviewer: ```suggestion
#### Types of validation

```

Use a combination of client side and server side validation for the best user experience. Catching basic errors with client side validation allows the user to quickly resolve the error **before** submitting the form. 

##### Client side validation
Client side validation, sometimes also referred to as inline validation, is an initial check that happens in the browser to ensure required fields are filled out and that the value is in the correct format. Learn more about [client side validation](https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation).

<!-- TODO uncomment example when form primitives is merged up -->
<!-- ![Example of client side form validation](/assets/components/form/primitives/form-validation-client.png) -->

> ##### Server side validation

Reviewer: ```suggestion
##### Server side validation

```
